relay/sources/rudder-pkg/lib/rudder-pkg/rpkg.py

```diff
@@ -0,0 +1,151 @@
+import os, logging, re, json
+from pkg_resources import parse_version
+
+import rudderPkgUtils as utils
+
+"""
+    Versions can contains the words "-SNAPSHOT" at the end, this should not be a problem at the moment since it is only present in nightly.
+    Moreover, we should not have to compare a nightly version to a release one.
+"""
+class Version:
+    def __init__(self, pluginLongVersion):
+        match = re.search(r'(?P<rudderVersion>[0-9]+\.[0-9]+)-(?P<pluginShortVersion>[0-9]+\.[0-9]+(-(?P<mode>[a-zA-Z]+)){0,1})', pluginLongVersion)
```

Comment:
? is better than {0,1}

Reply:
i'm pretty sure that pluginShortVersion should not include mode

Reply:
shouldn't it be a match instead of a search ? (or havr ^ and $)

Reply:
True, for 1 and 2.
For the search vs match, I do not see why we would anchor the regex since we only want one match in the entire string.